pyfr/__main__.py

```diff
@@ -201,83 +200,82 @@ def process_partition_info(args):
+        else:
+            pwts = [1]*int(args.np)
 
-    # Get the partitioning name
-    pname = args.name or str(len(pwts))
-    if not re.match(r'\w+$', pname):
-        raise ValueError('Invalid partitioning name')
+        # Element weights
+        if args.elewts == ['balanced']:
+            ewts = None
+        elif len(etypes) == 1:
+            ewts = {etypes[0]: 1}
+        else:
+            ewts = {e: int(w) for e, w in (ew.split(':') for ew in args.elewts)}
 
-    # Check it does not already exist unless --force is given
-    if pname in mesh['partitionings'] and not args.force:
-        raise ValueError('Partitioning already exists; use -f to replace')
+        # Ensure all weights have been provided
+        if ewts is not None and len(ewts) != len(etypes):
+            missing = ', '.join(set(etypes) - set(ewts))
+            raise ValueError(f'Missing element weights for: {missing}')
 
-    # Path to store the partitioning in the mesh
-    ppath = f'partitionings/{pname}'
+        # Get the partitioning name
+        pname = args.name or str(len(pwts))
+        if not re.match(r'\w+$', pname):
+            raise ValueError('Invalid partitioning name')
 
-    # Partitioner-specific options
-    opts = dict(s.split(':', 1) for s in args.popts)
+        # Check it does not already exist unless --force is given
+        if pname in mesh['partitionings'] and not args.force:
+            raise ValueError('Partitioning already exists; use -f to replace')
 
-    # Create the partitioner
-    if args.partitioner:
-        part = get_partitioner(args.partitioner, pwts, ewts, opts=opts)
-    else:
-        for name in sorted(cls.name for cls in subclasses(BasePartitioner)):
-            try:
-                part = get_partitioner(name, pwts, ewts)
-                break
-            except OSError:
-                pass
-        else:
-            raise RuntimeError('No partitioners available')
+        # Path to store the partitioning in the mesh
+        ppath = f'partitionings/{pname}'
 
-    # Partition the mesh
-    pinfo = part.partition(mesh, args.progress)
-    (partitioning, pregions), (neighbours, nregions) = pinfo
+        # Partitioner-specific options
+        opts = dict(s.split(':', 1) for s in args.popts)
 
-    # Write out the new partitioning
-    with args.progress.start('Write partitioning'):
-        if ppath in mesh:
-            mesh[f'{ppath}/eles'][:] = partitioning
-            del mesh[f'{ppath}/neighbours']
+        # Create the partitioner
+        if args.partitioner:
+            part = get_partitioner(args.partitioner, pwts, ewts, opts=opts)
         else:
-            mesh[f'{ppath}/eles'] = partitioning
-
-        mesh[f'{ppath}/neighbours'] = neighbours
-        mesh[f'{ppath}/eles'].attrs['regions'] = pregions
-        mesh[f'{ppath}/neighbours'].attrs['regions'] = nregions
+            for name in sorted(cls.name for cls in subclasses(BasePartitioner)):
```

Comment:
Do you have a preferred auto-formatting tool (Black, Ruff, etc.) to delegate this kind of thing to?

Reply:
I've not had much luck with auto-formatters but Ruff works great as a linter.  Two warnings you need to ignore are those related to our one-line classes such as `class CUDAError(Exception): pass` which we literally have 100's of in the code (it would prefer to add in four lines of extra whitespace for these) and our 'unused' imports (such as in `pyfr.backends.cuda.__init__.py` where we have `from pyfr.backends.cuda.base import CUDABackend` which is unused in the file itself but works nicely for hiding the internal structure of the backend from those who use it).